index.js

```diff
@@ -153,38 +194,56 @@ const OptionsMethod = (prefix) => ({
     ApiKeyRequired: false,
     AuthorizationType: 'None',
     HttpMethod: 'OPTIONS',
-    MethodResponses: [
-      {
-        StatusCode: 200,
-        ResponseModels: {
-          'application/json': 'Empty'
-        },
-        ResponseParameters: {
-          'method.response.header.Access-Control-Allow-Headers': true,
-          'method.response.header.Access-Control-Allow-Methods': true,
-          'method.response.header.Access-Control-Allow-Origin': true
-        }
-      }
-    ],
     Integration: {
-      Type: 'MOCK',
-      IntegrationResponses: [
+      Type: 'AWS_PROXY',
+      IntegrationHttpMethod: 'POST',
+      Uri: cf.sub(`arn:aws:apigateway:\${AWS::Region}:lambda:path/2015-03-31/functions/\${${prefix}PassthroughFunction.Arn}/invocations`)
+    }
+  }
+});
+
+const PassthroughFunctionRole = (lambda, prefix) => ({
+  Type: 'AWS::IAM::Role',
+  Properties: {
+    AssumeRolePolicyDocument: {
+      Statement: [
         {
-          StatusCode: 200,
-          ResponseParameters: {
-            'method.response.header.Access-Control-Allow-Headers': '\'Content-Type\'',
-            'method.response.header.Access-Control-Allow-Methods': '\'POST,OPTIONS\'',
-            'method.response.header.Access-Control-Allow-Origin': '\'*\''
+          Sid: 'passthroughrole',
+          Effect: 'Allow',
+          Principal: {
+            Service: 'lambda.amazonaws.com'
           },
-          ResponseTemplates: {
-            'application/json': '{}'
-          }
+          Action: 'sts:AssumeRole'
+        }
+      ]
+    },
+    Policies: [
+      {
+        PolicyName: `${prefix}PassthroughPolicy`,
+        PolicyDocument: {
+          Statement: [
+            {
+              Effect: 'Allow',
+              Action: [
+                'logs:*'
+              ],
+              Resource: [
+                'arn:aws:logs:*:*:*'
+              ]
+            },
+            {
+              Effect: 'Allow',
+              Action: [
+                'lambda:InvokeFunction'
+              ],
+              Resource: [
+                cf.getAtt(lambda, 'Arn')
+              ]
```

Comment:
This doesn't seem to be working as I'd hoped. The `WebhookPassthroughFunction` Lambda throws an error saying this role is not authorized to perform: lambda:InvokeFunction on my Lambda function. Any suggestions?

Reply:
It is the string that I pass into `hookshot.passthrough(HERE)`. I believe that is `The logical resource name of the hookshot caller's lambda`.

A sanitized version of my error message looks like this:
`"User: arn:aws:sts::1234:assumed-role/stack-name-WebhookPassthroughFunctionRo-1234/stack-name-WebhookPassthroughFunction-1234 is not authorized to perform: lambda:InvokeFunction on resource: arn:aws:lambda:us-east-1:1234:function:MyLambdaFunctionName"`